### ISSUE TITLE
Universal user agent not working in next.js edge runtime

### ISSUE THREAD

**Issue by Perfect7M** [gr2m/universal-user-agent#77]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `universal-user-agent@6.0.0` for the project I'm working on.

Universal user agent is used by [Octokit](https://github.com/octokit/octokit.js) but when importing it in next.js edge runtime fails. It seems that there `provess.version` is `undefined`. Following code works and resolve issue with using Ocotkit there, but it can be not sufficient. Feel free to let me know how this should be solved, and I'm happy to help on fix. 

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/universal-user-agent/dist-web/index.js b/node_modules/universal-user-agent/dist-web/index.js
index c550c02..a89b11f 100644
--- a/node_modules/universal-user-agent/dist-web/index.js
+++ b/node_modules/universal-user-agent/dist-web/index.js
@@ -3,7 +3,7 @@ function getUserAgent() {
         return navigator.userAgent;
     }
     if (typeof process === "object" && "version" in process) {
-        return `Node.js/${process.version.substr(1)} (${process.platform}; ${process.arch})`;
+        return `Node.js/${process.version?.substr(1)} (${process.platform}; ${process.arch})`;
     }
     return "<environment undetectable>";
 }
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by gr2m** [comment]
very odd. I'd like to

1. understand under what circumstances `process.version` is undefined
2. add a comment to explain said circumstance
3. fallback to something more useful than `Node.js/undefined`, maybe `Node.js/<unknown version>`

I'm open to a pull request for that. But right now I feel like this is a problem that should be fixed upstream in next.js, not here.

**Comment by Perfect7M** [comment]
Hi Gregor, sorry I bothered you. 

I reported cross issue on next.js platform - https://github.com/vercel/next.js/issues/54739
I agree that it should be resolved there, and it's a problem with Vercel edge runtime that is incompatible with nodejs.

To answer your questions and give you more context:
Vercel is making its "edge runtime" which is not compatible with nodejs. I think they implement some sort of `process` properties, and `process` global var itself but not the `process.version`
Here is an example of someone issue: https://github.com/vercel/next.js/issues/33122

For some reason `"version" in process` is `true` but `process.version` is still `undefined`. Seems that value is assigned with `undefined` on purpose.

<img width="278" alt="image" src="https://github.com/gr2m/universal-user-agent/assets/5896181/3649e352-5fd6-4e9f-a232-73895ae3c440">


Solution should be different from my code - more strict check for `process.version` being a string.

Feel free to close this ticket. I find this issue and someone might stumble upon it too, made it just for visibility.
